### ISSUE TITLE
implement deferred loading in fasta

### ISSUE THREAD

**Issue by sigmundch** [dart-lang/sdk#30273]

**Comment by dgrove** [comment]
@sigmundch any updates on this?

**Comment by sigmundch** [comment]
Just updated the comment above: we are now passing dart2js unit tests regarding correct splitting of deferred units. And I've validated that the output manually for a few small examples.

Next - I'll be validating them on a large app and move on to error checking.

**Comment by sigmundch** [comment]
quick update: runtime checks for using deferred elements before they are loaded are now inserted.

We are missing still compile-time errors (e.g. when extending a deferred class), which need to be handled likely in the outline builder stage. However, we reached the point where we have all we need to unblock the release of dart2js with kernel at this time, so I lowered the priority to P2 and unassigned myself for now. I may be able to help implement that last part if needed in the future, just wanted to make it clear that I'm not actively pursuing it at this moment.

**Comment by dgrove** [comment]
@sigmundch any updates on this? Can we defer the remaining work past Dart2Stable?

**Comment by sigmundch** [comment]
No updates from my end - I'm no longer actively working on it and left it for the FE team to do the final steps.

Missing features are about statically checking for certain kind of errors, so if we don't want to introduce new errors past Dart2Stable, then it might be relevant to do it sooner. If we are ok reintroducing errors in a later release, then it can be done after Dart2Stable.

**Comment by dgrove** [comment]
@sigmundch do you know if these errors are produced as expected by the analyzer?

**Comment by sigmundch** [comment]
they are

**Comment by dgrove** [comment]
In that case, we will very likely defer this past Dart2Stable.

**Comment by dgrove** [comment]
Moved to Dart 2.1

**Comment by kmillikin** [comment]
I will implement the missing static checks for deferred types.

**Comment by kmillikin** [comment]
This is not a feature that is connected to the Dart 2.1 release.
